Spigot-API-Patches/0232-added-PlayerEntityReceiveEvent.patch

```diff
@@ -0,0 +1,84 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: ysl3000 <yannicklamprecht@live.de>
+Date: Mon, 12 Oct 2020 23:06:57 +0200
+Subject: [PATCH] added PlayerEntityReceiveEvent
+
+
+diff --git a/src/main/java/io/papermc/paper/event/packet/PlayerEntityReceiveEvent.java b/src/main/java/io/papermc/paper/event/packet/PlayerEntityReceiveEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..9618b032031920c722b36f59bfbc3d6042d920fb
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/packet/PlayerEntityReceiveEvent.java
+@@ -0,0 +1,72 @@
++package io.papermc.paper.event.packet;
++
++import org.bukkit.entity.Entity;
++import org.bukkit.entity.Player;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.Event;
++import org.bukkit.event.HandlerList;
++import org.jetbrains.annotations.NotNull;
++
++/**
++ * Is called before a {@link Player} receives a {@link Entity}.
++ *
++ * E.g. can be used for the following:
++ *
++ * <ul>
++ *     <li>Hiding all kind of entities by calling {@link Cancellable#setCancelled(boolean)} with value true</li>
++ *     <li>Disguising entities as others e.g. player npc.
++ *         Cancel the event and replace it with a playerpacket with the same entity id.
++ *     </li>
++ * </ul>
++ */
++public class PlayerEntityReceiveEvent extends Event implements Cancellable {
```

Comment:
extends PlayerEvent

Reply:
we should also have an untrack event, so maybe this should be PlayerTrackEntityEvent and the other PlayerUntrackEntityEvent

Reply:
done separate.